src/components/TransactionsTab/TransactionsTab.tsx

```diff
@@ -109,86 +116,123 @@ const TransactionsTab = ({
-                  tokens={tokens}
-                  chainId={chainId!}
-                  key={`${transaction.hash}-${transaction.nonce}`}
-                />
-              ))
-            ) : (
-              <NoTransactions>
-                <IconContainer>
-                  <Icon name="transaction" />
-                </IconContainer>
-                {t("wallet:noActiveTransactions")}
-              </NoTransactions>
-            )}
-          </TransactionContainer>
-          {completedTransactions && (
+    <AnimatePresence>
+      {open && (
+        <Container
+          ref={containerRef}
+          animate={{ x: 0 }}
+          transition={{ duration: 0.3 }}
+          initial={{ x: "24rem" }}
+          exit={{ x: "24rem" }}
+        >
+          <BackButton
+            animate={{ y: 0 }}
+            transition={{ duration: 0.5 }}
+            initial={{ y: "-5rem" }}
+            exit={{ opacity: 0 }}
+            onClick={() => setTransactionsTabOpen(false)}
+          >
+            <Icon name="chevron-right" iconSize={1} />
+          </BackButton>
+          <WalletHeader>
+            <NetworkInfoContainer>
+              <NetworkName>
+                {addressMapping[chainId] || t("wallet:unsupported")}
+              </NetworkName>
+              <Balances>{formatUnits(balance).substring(0, 5)} ETH</Balances>
+            </NetworkInfoContainer>
+            <WalletAnchorTag
+              target="_blank"
+              rel="noreferrer"
+              href={
+                chainId
+                  ? `https://${hrefAddressMapping[chainId]}etherscan.io/address/${address}`
+                  : ""
+              }
+            >
+              <ConnectionStatusCircle $connected={!!address} />
+              <InfoHeading>
+                {isUnsupportedNetwork
+                  ? t("wallet:unsupported")
+                  : addressOrName
+                  ? addressOrName
+                  : t("wallet:notConnected")}
+              </InfoHeading>
+            </WalletAnchorTag>
+          </WalletHeader>
+
+          <TransactionsContainer
+            ref={transactionsScrollRef}
+            $overflow={overflow}
+          >
             <Legend>
-              <LegendLine>{t("wallet:completedTransactions")}</LegendLine>
+              <LegendLine>
+                {t("wallet:activeTransactions").toUpperCase()}
+              </LegendLine>
             </Legend>
-          )}
-          <TransactionContainer>
-            {completedTransactions.length > 0 ? (
-              completedTransactions
-                .slice(0, 10)
-                .map((transaction) => (
+            <TransactionContainer>
+              {pendingTransactions.length ? (
+                pendingTransactions.map((transaction) => (
                   <WalletTransaction
                     transaction={transaction}
                     tokens={tokens}
                     chainId={chainId!}
                     key={transaction.hash}
```

Comment:
not all transacitons in this list will have a hash. Suggest:

```
{`${transaction.hash}-${transaction.nonce}-${transaction.expiry}`}
```

Same for below.